google-cloud-firestore/lib/google/cloud/firestore/convert.rb

```diff
@@ -0,0 +1,545 @@
+              Integer value.integer_value
+            when :double_value
+              value.double_value
+            when :timestamp_value
+              timestamp_to_time value.timestamp_value
+            when :string_value
+              value.string_value
+            when :bytes_value
+              StringIO.new Base64.decode64 value.bytes_value
+            when :reference_value
+              Google::Cloud::Firestore::Document.from_path \
+                value.reference_value, context
+            when :geo_point_value
+              value.geo_point_value.to_hash
+            when :array_value
+              value.array_value.values.map { |v| value_to_raw v, context }
+            when :map_value
+              fields_to_hash value.map_value.fields, context
+            end
+          end
+
+          def raw_to_value obj
+            if NilClass === obj
+              Google::Firestore::V1beta1::Value.new null_value: :NULL_VALUE
+            elsif TrueClass === obj || FalseClass === obj
+              Google::Firestore::V1beta1::Value.new boolean_value: obj
+            elsif Integer === obj
+              Google::Firestore::V1beta1::Value.new integer_value: obj
+            elsif Numeric === obj # Any number not an integer is a double
+              Google::Firestore::V1beta1::Value.new double_value: obj.to_f
+            elsif Time === obj || DateTime === obj || Date === obj
+              Google::Firestore::V1beta1::Value.new \
+                timestamp_value: time_to_timestamp(obj.to_time)
+            elsif String === obj || Symbol === obj
+              Google::Firestore::V1beta1::Value.new string_value: obj.to_s
+            elsif Google::Cloud::Firestore::Document::Reference === obj
+              Google::Firestore::V1beta1::Value.new reference_value: obj.path
+            elsif Array === obj
+              values = obj.map { |o| raw_to_value(o) }
+              Google::Firestore::V1beta1::Value.new(array_value:
+                Google::Firestore::V1beta1::ArrayValue.new(values: values))
+            elsif Hash === obj
+              if obj.keys.sort == [:latitude, :longitude]
+                Google::Firestore::V1beta1::Value.new(geo_point_value:
+                  Google::Type::LatLng.new(obj))
+              else
+                fields = hash_to_fields obj
+                Google::Firestore::V1beta1::Value.new(map_value:
+                  Google::Firestore::V1beta1::MapValue.new(fields: fields))
+              end
+            elsif obj.respond_to?(:read) && obj.respond_to?(:rewind)
+              obj.rewind
+              content = obj.read.force_encoding "ASCII-8BIT"
+              encoded_content = Base64.strict_encode64 content
+              Google::Firestore::V1beta1::Value.new bytes_value: encoded_content
+            else
+              fail ArgumentError,
+                   "A value of type #{obj.class} is not supported."
+            end
+          end
+
+          def create_writes doc_path, data
```

Comment:
I got confused by this name. This is more of a write_for_create than a createWrite(). The last one suggest that it adds a new write, but doesn't necessarily say that it is about the 'create' operation.

Reply:
These methods are all internal private helper methods, so the names aren't as important as they aren't part of the public API. It can return more than one Write value, which is why its plural. We can rename these private helper methods if that is wanted.